src/Stryker.Core/Stryker.Core.UnitTest/MutationTest/MutationTestProcessTests.cs

```diff
@@ -288,5 +288,155 @@ public void MutationTestProcess_ShouldCallExecutorForEveryMutant()
             reporterMock.Verify(x => x.OnMutantTested(mutant), Times.Once);
             reporterMock.Verify(x => x.OnAllMutantsTested(It.IsAny<ProjectComponent>()), Times.Once);
         }
+
+        [Fact]
+        public void MutationTestProcess_ShouldNotTest_WhenAllMutationsWereSkipped()
+        {
+            var mutant = new Mutant() { Id = 1, ResultStatus = MutantStatus.Skipped};
+            string basePath = Path.Combine(_filesystemRoot, "ExampleProject.Test");
+            var input = new MutationTestInput()
+            {
+                ProjectInfo = new Core.Initialisation.ProjectInfo()
+                {
+                    TestProjectPath = basePath,
+                    ProjectContents = new FolderComposite()
+                    {
+                        Name = "ProjectRoot",
+                        Children = new Collection<ProjectComponent>() {
+                        new FileLeaf() {
+                            Name = "SomeFile.cs",
+                            Mutants = new Collection<Mutant>() { mutant }
+                        }
+                    }
+                    },
+                    ProjectUnderTestAssemblyName = "ExampleProject",
+                    ProjectUnderTestPath = Path.Combine(_filesystemRoot, "ExampleProject"),
+                    TargetFramework = "netcoreapp2.0",
+                },
+                AssemblyReferences = new ReferenceProvider().GetReferencedAssemblies()
+            };
+            var reporterMock = new Mock<IReporter>(MockBehavior.Strict);
+            reporterMock.Setup(x => x.OnMutantTested(It.IsAny<Mutant>()));
+            reporterMock.Setup(x => x.OnAllMutantsTested(It.IsAny<ProjectComponent>()));
+            reporterMock.Setup(x => x.OnStartMutantTestRun(It.IsAny<IList<Mutant>>()));
+
+            var executorMock = new Mock<IMutationTestExecutor>(MockBehavior.Strict);
+            executorMock.SetupGet(x => x.TestRunner).Returns(Mock.Of<ITestRunner>());
+            executorMock.Setup(x => x.Test(It.IsAny<Mutant>()));
+
+            var options = new StrykerOptions(fileSystem: new MockFileSystem(), basePath: basePath);
+
+            var target = new MutationTestProcess(input,
+                reporterMock.Object,
+                executorMock.Object);
+
+            var testResult = target.Test(options);
+
+            executorMock.Verify(x => x.Test(mutant), Times.Never);
+            reporterMock.Verify(x => x.OnStartMutantTestRun(It.Is<IList<Mutant>>(y => y.Count == 1)), Times.Never);
+            reporterMock.Verify(x => x.OnMutantTested(mutant), Times.Never);
+            reporterMock.Verify(x => x.OnAllMutantsTested(It.IsAny<ProjectComponent>()), Times.Never);
+            Assert.Null(testResult.MutationScore);
```

Comment:
We try to use shouldly for assertions. This would look like:
```suggestion
            testResult.MutationScore.ShouldBeNull();
```